### PR TITLE
Update Heasoft to 6.31.1 starting from ubuntu 22.04

### PR DESCRIPTION
**Pull request by zoghbi-a**
- Re-organized the build script to read the version numbers from a single json file for easy future updates.
- The base images use sciserver-essentials/3.0 (sciserver-base and sciserver-jupyter), which start from ubuntu 22.04
- Removed the heasoft version number from the name of the final folder.
- Some additional tweaks to reduce the size of the final image.
- Added a test folder with instruction on running minimal tests that ensures the software are installed correctly. The tests can be run both in a local build after downloading testing data as well as on sciserver.

### CODE REVIEW

File: heasarc/README.md [hunk 1]
```diff
@@ -1,92 +1,125 @@
...
+cd data/; ln /FTP FTP; cd ..
+bash run_tests.sh
+```
 
-- Create a new conda environment: ciao
 
-- Install ciao and related packages using conda (or mamba) using the relevant channel.
 
+### Description Of The Images
+In the folloing, each image starts with the image before it.
```
Comment (esabol) [inline heasarc/README.md:65]: Typo on this line.

File: heasarc/heasarc/entry.sh [hunk 2]
```diff
@@ -5,4 +5,4 @@
 if ! [ -t 0 ] ; then
     source ~/.bashrc
 fi
-exec "$@"
+exec "$@"
```
Comment (esabol) [inline heasarc/heasarc/entry.sh:8]: Missing carriage return (linefeed, technically) at the end of this line.

File: heasarc/heasarc/kernel.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    "argv": [
+        "conda", "run", "--no-capture-output", "-n", "heasoft", 
+        "python", "-m", "ipykernel_launcher", "-f", "{connection_file}"
+    ],
+    "display_name": "(heasoft)",
+    "language": "python",
+    "metadata": {
+        "debugger": true
+    }
+}
```
Comment (esabol) [inline heasarc/heasarc/kernel.json:11]: Missing carriage return (linefeed, technically) at the end of this line.

File: heasarc/xmmsas/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+source \$SAS_DIR/setsas.csh \n\
+if ! [ -d $headata ]; then \n \
+   echo '\\n** HEASARC data Volume is needed for SAS_CCFPATH. Please mounted it when creating the container **\\n' \n\
+fi \n\
+" > miniconda3/envs/${pythonenv}/etc/conda/activate.d/activate_xmmsas.csh
+## ------------------- ##
+
+
+# reset 
+WORKDIR /home/${user}
```
Comment (esabol) [inline heasarc/xmmsas/Dockerfile:82]: Missing carriage return (linefeed, technically) at the end of this line.

**Review by esabol — CHANGES_REQUESTED** [review]

**Review by esabol — APPROVED** [review]

**Comment by zoghbi-a** [comment]
[fda4110](https://github.com/sciserver/sciserver-compute-images/pull/20/commits/fda4110329d7beed0fdd03d692e832a80b122b78) foces numpy=1.23 in ciao. The recent numpy=1.24 release breaks some tasks, so until the ciao team fixes that, downgrading numpy works for now.

**Pull request merged**